pwnlib/rop/rop.py

```diff
@@ -1341,6 +1370,19 @@ def __getattr__(self, attr):
                 count = int(attr.split('_')[1])
             return self.search(move=count)
 
+        #
+        # Check for 'jmp_esp' or 'jmp_rsp'
+        #
+        sp_regs = ['esp', 'rsp']
+
+        if attr.startswith('jmp') \
+        and attr.split('_')[1] in sp_regs:
```

Comment:
This seems really error prone, we should just use `ELF.search(..., executable=True)` for the specific byte sequence -- especially since it doesn't change between architectures.

Reply:
My PR loads all `ROP.jmp_$sp` addresses as gadgets at `__load`, filtering out based on `self._badchars`.
If you think it is better, I move the code to `__getattr__`. It will only return the first match that does not have any badchar. Using this approach, I do not need to touch `self.gadgets` while loading.

Reply:
I'd rather not filter out JMP gadgets and just search for the specific byte sequence, since this is special-case code anyway.

Maybe in a future PR we can search for all `jmp reg` instances and put them in e.g. `ROP.jmp_gadgets`

For now, let's hard-code the bytes for `jmp $sp` and search with `ELF.search`

Reply:
I've moved the code to `__getattr__`, used the hard-coded opcodes to search with `ELF.search` until the gadget address does not have any badchar.